src/solver/impls/slepc/slepc.cxx

```diff
@@ -36,6 +36,20 @@
 
 #include <cstdlib>
 
+#if BOUT_USE_SIGFPE
+#include <fenv.h>
+#endif
+
+namespace {
+/// Disable floating-point exceptions in a scope, reenable them on exit
+struct QuietFPE {
+#if BOUT_USE_SIGFPE
+  QuietFPE() { fedisableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW); }
+  ~QuietFPE() { feenableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW); }
```

Comment:
Is it possible to store the state of the flags?

Otherwise two nested regions of QuietFPE would produce difficult to debug behaviour, as the flags are disabled, but re-enabled to early.

I am not sure FPE handling is working at all right now ...

Reply:
Good point. I think `fegetexcept` looks like it does this, will check.

I'm not really sure if the FPE handling is working correctly either. I was trying to work around some bits using `std::isnan`, but even that was triggering signals.